File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Task/Domain/TaskHeaderProtocol.vb
Public Class TaskHeaderProtocol
    Inherits DomainObject(Of Integer)
    Implements IAuditable

#Region " Properties "

    Public Overridable Property Year As Short?

    Public Overridable Property Number As Integer?

    Public Overridable Property Header As TaskHeader

    Public Overridable Property Protocol As Protocol

    Public Overridable Property LastChangedDate As DateTimeOffset? Implements IAuditable.LastChangedDate

    Public Overridable Property LastChangedUser As String Implements IAuditable.LastChangedUser

    Public Overridable Property RegistrationDate As DateTimeOffset Implements IAuditable.RegistrationDate

    Public Overridable Property RegistrationUser As String Implements IAuditable.RegistrationUser

#End Region

End Class

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Tasks/TaskHeaderProtocol.cs
using System;
using VecompSoftware.DocSuiteWeb.Entity.Protocols;

namespace VecompSoftware.DocSuiteWeb.Entity.Tasks
{
    public class TaskHeaderProtocol : DSWBaseEntity
    {
        #region [ Constructor ]
        public TaskHeaderProtocol() : this(Guid.NewGuid()) { }

        public TaskHeaderProtocol(Guid uniqueId)
            : base(uniqueId)
        {

        }
        #endregion

        #region [ Properties ]
        public short? Year { get; set; }
        public int? Number { get; set; }
        #endregion

        #region [ Navigation Properties ]
        public virtual TaskHeader TaskHeader { get; set; }
        public virtual Protocol Protocol { get; set; }
        #endregion
    }
}
